server/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -760,6 +771,8 @@ public ClusterState execute(BatchExecutionContext<StartedShardUpdateTask> batchE
                     maybeUpdatedState = ClusterState.builder(maybeUpdatedState).metadata(metadataBuilder).build();
                 }
 
+                maybeUpdatedState = maybeRemoveIndexRefreshBlocks(maybeUpdatedState, indicesWithUnpromotableShardsStarted);
+
                 assert assertStartedIndicesHaveCompleteTimestampRanges(maybeUpdatedState);
```

Comment:
Maybe we could assert that we never see an index with a refresh block and started search shards here?

Reply:
Done in https://github.com/elastic/elasticsearch/pull/120807/commits/b766d8e7f603de554ab684c3123a2a2a26196562